.setup/bin/populate_office_hours_queue.py

```diff
@@ -0,0 +1,191 @@
+        queue_entry["name"] = name_lookup[queue_entry["user_id"]]
+
+        # get a random join queue time
+        time_in = random.randint(300, 600)
+        queue_entry["time_in"] = datetime.now() - timedelta(seconds = time_in)
+
+        # get a random exit queue time
+        time_out = random.randint(0, 300)
+        if queue_entry["current_state"] == "done":
+            queue_entry["time_out"] = datetime.now() - timedelta(seconds = time_out)
+        else:
+            queue_entry["time_out"] = None
+
+        queue_entry["added_by"] = queue_entry["user_id"]
+
+        # find helper if the student is currently helped
+        if queue_entry["current_state"] == "waiting" or queue_entry["removal_type"] == "self":
+            queue_entry["help_started_by"] = None
+        else:
+            queue_entry["help_started_by"] = random.choice(all_grader_ids)
+
+        if queue_entry["removal_type"] == None:
+            queue_entry["removed_by"] = None
+        elif queue_entry["removal_type"] == "helped":
+            # usually the grader who started will finish helping
+            if random.random() < 0.9:
+                queue_entry["removed_by"] = queue_entry["help_started_by"]
+            else:
+                queue_entry["removed_by"] = random.choice(all_grader_ids)
+        elif queue_entry["removal_type"] == "self" or queue_entry["removal_type"] == "self_helped":
+            queue_entry["removed_by"] = queue_entry["user_id"]
+        else:
+            # student is being emptied/removed out of the queue
+            queue_entry["removed_by"] = random.choice(all_grader_ids)
+
+        # contact information
+        if queues_lookup[queue_entry["queue_code"]]["contact_information"]:
+            queue_entry["contact_info"] = queue_entry["user_id"] + "@sample.com"
+        else:
+            queue_entry["contact_info"] = None
+
+        res = conn.execute("SELECT max(time_in) FROM queue WHERE user_id = '{}' AND UPPER(TRIM(queue_code)) = UPPER(TRIM('{}')) AND (removal_type IN ('helped', 'self_helped') OR help_started_by IS NOT NULL)".format(queue_entry["user_id"], queue_entry["queue_code"]))
+        queue_entry["last_time_in_queue"] = res.fetchall()[0][0]
+        res.close()
+
+        if queue_entry["help_started_by"] != None:
+            time_start = random.randint(time_out, time_in)
+            queue_entry["time_help_start"] = datetime.now() - timedelta(seconds = time_start)
+        else:
+            queue_entry["time_help_start"] = None
+        
+        # people usually won't be pausing
+        if random.random() < 0.8:
+            queue_entry["paused"] = False
+            queue_entry["time_paused"] = 0
+            queue_entry["time_paused_start"] = None
+        # this user is pausing or has paused
+        else:
+            # currently pausing
+            queue_entry["paused"] = random.choice([True, False])
+            queue_entry["time_paused"] = random.choice([0, random.randint(0, time_in)])
```

Comment:
Why do we need the `random.choice()` here?  Do we want there to be a 50% chance that the time paused is 0?  Just to be clear, I'm just wondering about the logic here, not necessarily requesting changes.

Reply:
I believe I added the `random.choice()` here because I think it's more likely that a user haven't paused before and having it as `random.randint(0, time_in)` would give less "realistic" data. (However, based on testing I think the pausing time is not working as I intend so code around this area might be changed)